java/feathub-connectors/flink-connectors/flink-connector-redis/src/main/java/com/alibaba/feathub/flink/connectors/redis/sink/JedisClusterClient.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 The FeatHub Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.connectors.redis.sink;
+
+import org.apache.flink.util.Preconditions;
+
+import org.apache.commons.collections.BidiMap;
+import org.apache.commons.collections.bidimap.DualHashBidiMap;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.exceptions.JedisNoScriptException;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A subclass of {@link JedisClient} that connects to a Redis cluster. Used for Redis Cluster mode.
+ */
+public class JedisClusterClient implements JedisClient {
+    private static final byte[] sampleKey = "sampleKey".getBytes();
+
+    private final JedisCluster cluster;
+
+    private final BidiMap scriptShaMap;
```

Comment:
Would it be simpler to use two hash maps instead of introducing the extra dependency on `commons-pool2`? It is preferred to keep the connector jar size as small as possible.

Reply:
It is `JedisCluster` instead of `BidiMap` that is relying on `commons-pool2`, so replacing `BidiMap` with simpler hash maps could not avoid introducing this dependency.

Reply:
Sounds good.